pycbc/events/stat.py

```diff
@@ -1758,6 +1757,155 @@ def coinc_lim_for_thresh(self, s, thresh, limifo,
         return loglr
 
 
+
+class ExpFitFgBgKDEStatistic(ExpFitFgBgNormStatistic):
+    """
+    The ExpFitFgBgNormStatistic with a mass  and spin weighting factor determined by KDE
+
+    This is the same as the ExpFitFgBgNormStatistic except the likelihood
+    is multiplied by the ratio of signal KDE to template KDE over template params.
+    """
+
+    def __init__(self, sngl_ranking, files=None, ifos=None, **kwargs):
+        """
+        Create a statistic class instance
+
+        Parameters
+        ----------
+        sngl_ranking: str
+            The name of the ranking to use for the single-detector triggers.
+        files: list of strs, needed here
+            A list containing the filenames of hdf format files used to help
+            construct the coincident statistics. The files must have a 'stat'
+            attribute which is used to associate them with the appropriate
+            statistic class.
+        ifos: list of strs, not used here
+            The list of detector names
+        """
+        ExpFitFgBgNormStatistic.__init__(self, sngl_ranking, files=files,
+                                         ifos=ifos, **kwargs)
+        parsed_attrs = [f.split('-') for f in self.files.keys()]
+        self.data_name = [at[0] for at in parsed_attrs if
+                       (len(at) == 2 and at[1] == 'kde_file')]
+        if not len(self.data_name):
+            raise RuntimeError("None of the statistic files has the required "
+                               "attribute called {dname}-kde_file !")
+        self.kde_by_tid = {}
+        for dname in self.data_name:
+            self.kde_by_tid[dname] = self.assign_kdes(dname)
+            #print('self.kde_by_tid=', self.kde_by_tid[dname])
+        self.curr_tnum = None       
+ 
+    def assign_kdes(self, dname):
+        """
+        Extract fits from fit files
+
+        Parameters
+        -----------
+        dname: str
+            signal or template.
+
+        Returns
+        -------
+        rate_dict: dict
```

Comment:
The return value is not called 'rate_dict' it is called 'kde_by_tid_dict', please edit the comment here

Reply:
Or you might not want to use a return value at all ..

Reply:
Or you might want to return just the kde values, not a dictionary, because it's not necessary to have such a complicated structure for the KDEs as it was for the background fits that depend on both the detector and the template